artcommon/artcommonlib/exectools.py

```
@@ -584,7 +749,13 @@ async def cmd_gather_async(cmd: Union[List[str], str], check: bool = True, **kwa
     proc = await asyncio.subprocess.create_subprocess_exec(cmd_list[0], *cmd_list[1:], **kwargs)
     span.set_attribute("process.pid", proc.pid)
 
-    stdout, stderr = await proc.communicate()
+    try:
+        stdout, stderr = await asyncio.wait_for(proc.communicate(), timeout=timeout)
+    except asyncio.TimeoutError:
+        logger.warning("Command timed out after %ss: %s", timeout, " ".join(cmd_list))
+        proc.kill()
+        await proc.wait()
+        raise ChildProcessError(f"Process {cmd_list!r} timed out after {timeout} seconds.")
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)